src/Components/Molecules/FMMapsTextField/index.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { memo } from "react";
+import Box from "@mui/material/Box";
+import Backdrop from "@mui/material/Backdrop";
+
+import { FAButton, FAIcon, FATextField, FATypography } from "@Atoms";
+
+import { DEFAULT_IMAGE_MAPS_PLACEHOLDER } from "@Definitions/constants";
+
+import { IFMMapsTextField } from "./FMMapsTextField.interface";
+
+export const FMMapsTextField = memo((props: IFMMapsTextField) => {
+  const {
+    testID,
+    label,
+    buttonLabel,
+    buttonIconName,
+    onButtonClick,
+    variant = "standard",
+    fullWidth = true,
+    ...textFieldProps
+  } = props;
+
+  return (
+    <Box data-testid={testID}>
+      <Box mb={0.5}>
+        <FATypography testID="typography-maps-label" variant="captionSemiBold">
+          {label}{" "}
+          {props.required && (
+            <FATypography
+              testID="typography-asterisk"
+              color="error.main"
+              component="span"
+            >
+              *
+            </FATypography>
+          )}
+        </FATypography>
+      </Box>
+      <Box position="relative">
+        <Backdrop open={true} sx={{ position: "absolute" }}>
+          <FAButton
+            testID="button-input-maps"
+            {...(buttonIconName && {
+              startIcon: (
+                <FAIcon testID="icon-button" color="inherit">
```

Comment:
icon class name in Figma match with filled icon, maybe you can add it here. [reference](https://fonts.google.com/icons?icon.query=location_on&icon.style=Filled)